docs/rfcs/header-component-api.md

```diff
@@ -0,0 +1,458 @@
+            />
+        </Navigation.PrimaryLinks>
+        <Navigation.SecondaryLinks>
+            <Navigation.SecondaryLinksGroup>
+                <Navigation.SecondaryLink
+                    label="News"
+                    href="https://www.theguardian.com/news"
+                />
+            </Navigation.SecondaryLinksGroup>
+            <Navigation.SecondaryLinksSupplementalLink
+                label="Masterclasses"
+                href="https://www.theguardian.com/guardian-masterclasses"
+            />
+        </Navigation.SecondaryLinks>
+    </Navigation>
+    <Subnav />
+</Header>
+```
+
+## Component breakdown
+
+### `<Header />`
+
+Generic header component. The logo is hard-coded.
+
+![Header](images/header-component-api/header.png)
+
+#### Example
+
+```jsx
+<Header>
+    <BrandBar />
+    <Navigation />
+    <Subnav />
+</Header>
+```
+
+### `<BrandBar />`
+
+The area at the top of the header that houses features related to the brand and customisation, rather than content discovery. Examples include the support message, top level links, the Guardian logo and the Edition Switch.
+
+The logo is hardcoded.
+
+![Brand bar](images/header-component-api/brand-bar.png)
+
+#### Example
+
+```jsx
+<BrandBar.Links>
+    <BrandBar.Link href icon={<Icon />} label until from />
+    <BrandBar.LinkList icon label until from>
+        <BrandBar.Link href label />
+    </BrandBar.LinkList>
+</BrandBar.Links>
+```
+
+#### `<BrandBar.Links />`
+
+A list of top-level links and menus that is displayed at the very top of the header.
+
+![Brand bar links](images/header-component-api/brand-bar-links.png)
+
+#### `<BrandBar.Link />`
+
+A link that is displayed in the `BrandBar.Links` section of the header, or within a `BrandBar.LinkList` dropdown.
+
+![Brand bar link](images/header-component-api/brand-bar-link.png)
+
+##### Props
+
+-   `href` _string_
+    -   navigation location when `BrandBar.Link` is clicked
+-   `onClick` _function_
+    -   logic that is executed when the `BrandBar.Link` is clicked
+-   `label` _string_
+    -   display label of the `BrandBar.Link`
+-   `icon` _ReactNode_
+    -   icon that appears alongside the label
+-   `from` _Breakpoint_
+    -   the breakpoint at which the `BrandBar.Link` becomes visible
+-   `until` _Breakpoint_
+    -   the breakpoint at which the `BrandBar.Link` is no longer displayed
+
+#### `<BrandBar.LinkList />`
+
+A dropdown menu that is displayed in the `BrandBar.Links` section of the header.
+
+![Brand bar link list](images/header-component-api/brand-bar-link-list.png)
+
+##### Props
+
+-   `label` _string_
+    -   display label of the `BrandBar.LinkList`
+-   `icon` _ReactNode_
+    -   icon that appears alongside the label
+-   `from` _Breakpoint_
+    -   the breakpoint at which the `BrandBar.LinkList` becomes visible
+-   `until` _Breakpoint_
+    -   the breakpoint at which the `BrandBar.LinkList` is no longer displayed
+
+### `<Navigation />`
```

Comment:
I wonder if it is worth using 'pillar' in the naming here? It seems integral but that could just be my perspective